src/ecs/resources/input/axis.rs

```diff
@@ -0,0 +1,11 @@
+use super::Button;
+
+/// Represents an axis made up of digital inputs, like W and S or A and D.
+/// Two of these could be analogous to a DPAD.
+#[derive(Serialize, Deserialize)]
+pub struct Axis {
+    /// Positive button, when pressed down axis value will return 1 if `neg` is not pressed down.
```

Comment:
@Xaeroxe Thoughts about my old design?  I was able to emulate every input device I came across with those 3 styles (the value types being `bool`, `float constraint`, `float unconstrained`).